File: rb/features.rb
# Answerbook SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module AnswerbookFeatures
  def self.make_feature(name)
    case name
    when "base"
      AnswerbookBaseFeature.new
    when "test"
      AnswerbookTestFeature.new
    else
      AnswerbookBaseFeature.new
    end
  end
end

File: py/features.py
# Answerbook SDK feature factory

from feature.base_feature import AnswerbookBaseFeature
from feature.test_feature import AnswerbookTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AnswerbookBaseFeature(),
        "test": lambda: AnswerbookTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
